src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs

```diff
@@ -138,14 +138,51 @@ public static IDataView LoadFeaturizedAdultDataset(MLContext mlContext)
                 .Append(mlContext.Transforms.Concatenate("Features", "workclass", "education", "marital-status",
                     "occupation", "relationship", "ethnicity", "native-country", "age", "education-num",
                     "capital-gain", "capital-loss", "hours-per-week"))
-                // Min-max normalized all the features
+                // Min-max normalize all the features
                 .Append(mlContext.Transforms.Normalize("Features"));
 
             var data = reader.Read(dataFile);
             var featurizedData = pipeline.Fit(data).Transform(data);
             return featurizedData;
         }
 
+        public static string DownloadMslrWeb10k()
+        {
+            var fileName = "MSLRWeb10KTrain720kRows.tsv";
+            if (!File.Exists(fileName))
+                Download("https://tlcresources.blob.core.windows.net/datasets/MSLR-WEB10K/MSLR-WEB10K_Fold1.TRAIN.500MB_720k-rows.tsv", fileName);
+            return fileName;
+        }
+
+        public static IDataView LoadFeaturizedMslrWeb10kDataset(MLContext mlContext)
+        {
+            // Download the training and validation files.
+            string dataFile = DownloadMslrWeb10k();
+
+            // Create the reader to read the data.
+            var reader = mlContext.Data.CreateTextLoader(
+                columns: new[]
+                {
+                    new TextLoader.Column("Label", DataKind.R4, 0),
+                    new TextLoader.Column("GroupId", DataKind.TX, 1),
+                    new TextLoader.Column("Features", DataKind.R4, new[] { new TextLoader.Range(2, 138) })
+                }
+            );
+
+            // Load the raw dataset.
+            var data = reader.Read(dataFile);
+
+            // Create the featurization pipeline. First, hash the GroupId column.
+            var pipeline = mlContext.Transforms.Conversion.Hash("GroupId")
```

Comment:
>mlContext.Transforms.Conversion.Hash("GroupId") [](start = 27, length = 47)

Doesn't TrainTestSplit take care of that?

Reply:
No, GroupId needs to be converted to Key first.

Reply:
https://github.com/dotnet/machinelearning/blob/3b9d407d9dc4f8c46fa85ab80575ef16d74df6df/src/Microsoft.ML.Data/TrainCatalog.cs#L201

Are you sure?

Reply:
I mean, I get that error when I try this pipeline without hashing GroupId